src/ir/effects.h

```diff
@@ -32,7 +32,7 @@ struct EffectAnalyzer
                  Expression* ast = nullptr)
     : ignoreImplicitTraps(passOptions.ignoreImplicitTraps),
       debugInfo(passOptions.debugInfo), features(features) {
-    if (ast) {
+    if (ast && !(ast->is<Const>() || ast->is<Nop>())) {
```

Comment:
This makes the code less readable, and I think this is unnecessary, unless you can show numbers that it really speeds things up.